packages/animations/lib/src/shared_z_axis_transition.dart

```diff
@@ -0,0 +1,394 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'utils/curves.dart';
+
+/// Used by [PageTransitionsTheme] to define a page route transition animation
+/// in which outgoing and incoming elements share a scale transition.
+///
+/// The shared axis pattern provides the transition animation between UI elements
+/// that have a spatial or navigational relationship. For example,
+/// transitioning from one page of a sign up page to the next one.
+///
+///
+/// The following example shows how the SharedZAxisPageTransitionsBuilder can
+/// be used in a [PageTransitionsTheme] to change the default transitions
+/// of [MaterialPageRoute]s.
+///
+/// ```dart
+/// MaterialApp(
+///   theme: ThemeData(
+///     pageTransitionsTheme: PageTransitionsTheme(
+///       builders: {
+///         TargetPlatform.android: SharedZAxisPageTransitionsBuilder(),
+///         TargetPlatform.iOS: SharedZAxisPageTransitionsBuilder(),
+///       },
+///     ),
+///   ),
+///   routes: {
+///     '/': (BuildContext context) {
+///       return Container(
+///         color: Colors.red,
+///         child: Center(
+///           child: MaterialButton(
```

Comment:
Please don't use MaterialButton (here and below), it's a leftover. RaisedButton, FlatButton, or OutlineButton would be OK.